script/bench.rb

```diff
@@ -49,13 +49,15 @@ def run(command, opt = nil)
       system(command, out: $stdout, err: :out)
     end
 
-  exit unless exit_status
+  abort("Command '#{command}' failed with exit status #{$?}") unless exit_status
 end
 
 begin
   require 'facter'
 rescue LoadError
   run "gem install facter"
+  # Facter requires CFPropertyList, but doesn't install it.
```

Comment:
The facter gem in 2.x fails when calling Facter.to_hash, because it `require`s 'cfpropertylist'. 

```
irb(main):001:0> require 'facter'
=> true
irb(main):002:0> Facter.to_hash
Error loading fact /Users/me/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/facter-2.4.6/lib/facter/ec2.rb: cannot load such file -- cfpropertylist
Error loading fact /Users/me/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/facter-2.4.6/lib/facter/macosx.rb: cannot load such file -- cfpropertylist
Error loading fact /Users/me/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/facter-2.4.6/lib/facter/processor.rb: cannot load such file -- cfpropertylist
LoadError: cannot load such file -- cfpropertylist
```

It looks like master on `facter` is a rewrite using C++ with ruby bindings, but on the 2.x branch, you can see the require here:

https://github.com/puppetlabs/facter/blob/2.x/lib/facter/util/macosx.rb#L9